test/framework/src/main/java/org/elasticsearch/transport/nio/TestingSocketEventHandler.java

```diff
@@ -19,38 +19,195 @@
 
 package org.elasticsearch.transport.nio;
 
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.message.ParameterizedMessage;
+import org.elasticsearch.nio.ChannelContext;
 import org.elasticsearch.nio.EventHandler;
 import org.elasticsearch.nio.NioSelector;
+import org.elasticsearch.nio.ServerChannelContext;
 import org.elasticsearch.nio.SocketChannelContext;
 
 import java.io.IOException;
 import java.util.Collections;
 import java.util.Set;
 import java.util.WeakHashMap;
+import java.util.concurrent.TimeUnit;
 import java.util.function.Consumer;
 import java.util.function.Supplier;
 
 public class TestingSocketEventHandler extends EventHandler {
 
-    private Set<SocketChannelContext> hasConnectedMap = Collections.newSetFromMap(new WeakHashMap<>());
+    private static final Logger logger = LogManager.getLogger(TestingSocketEventHandler.class);
 
-    public TestingSocketEventHandler(Consumer<Exception> exceptionHandler, Supplier<NioSelector> selectorSupplier) {
+    private final Set<SocketChannelContext> hasConnectedMap = Collections.newSetFromMap(new WeakHashMap<>());
+    private final Set<SocketChannelContext> hasConnectExceptionMap = Collections.newSetFromMap(new WeakHashMap<>());
+
+    TestingSocketEventHandler(Consumer<Exception> exceptionHandler, Supplier<NioSelector> selectorSupplier) {
         super(exceptionHandler, selectorSupplier);
     }
 
+    @Override
+    protected void acceptChannel(ServerChannelContext context) throws IOException {
+        long startTime = System.nanoTime();
```

Comment:
would it make sense to pass a time source to the constructor of TestingSocketEventHandler, so that we can pass in `threadPool.relativeTimeInMillis()`? This would allow the time to be mocked, and also avoid too many calls to `System.nanoTime`. Finally, it would also allow this new slow-logging functionality to be unit-tested

Reply:
I can make it configurable to test it. However, I do no understand the value of `threadPool.relativeTimeInMillis()`. This is only used in tests so I do not understand the the importance of avoiding calls to `System.nanotime`. Additionally, the time interval setting is set to tick every 200ms which means that this is going to trigger the log all the time.

Reply:
ok, fair point.